lib/bots/tobybot.ts

```diff
@@ -0,0 +1,194 @@
+            pem: pemString,
+            type: 'emitter'
+        });
+
+        // Throw if we didn't get either of the services.
+        if (!ghListener) {
+            throw new Error("Couldn't create a Github listener");
+        }
+        if (!ghEmitter) {
+            throw new Error("Couldn't create a Github emitter");
+        }
+        this.githubListenerName = ghListener.serviceName;
+        this.githubEmitterName = ghEmitter.serviceName;
+
+        // Github API handle
+        this.githubApi = (<GithubHandle>ghEmitter.apiHandle).github;
+        if (!this.githubApi) {
+            throw new Error('No Github API instance found');
+        }
+
+        // Worker methods, expandable
+        _.forEach([
+            {
+                events: [ 'issues' ],
+                listenerMethod: this.checkPalindrome,
+                name: 'checkPalindrom'
+            }
+        ], (reg: GithubRegistration) => {
+            ghListener.registerEvent(reg);
+        });
+    }
+
+    /**
+     * Checks a new or updated issue body.
+     * 1. Triggered by an `created` event on an Issue (which also happens when
+     *    the issue's body is updated)
+     * 2. Counts palindrome words in the issue body
+     * 3. Checks if the whole issue body is a palindrome
+     * 4. Add a comment on the issue with the palindrome count and whole text palindrome
+     *
+     * @param _registration GithubRegistration object used to register the method
+     * @param event         ServiceEvent containing the event information ('pull_request' event)
+     * @returns             A void Promise once execution has finished.
+     */
+    protected checkPalindrome = (_registration: GithubRegistration, event: ServiceEvent): Promise<void> => {
+        const issue = event.cookedEvent.data.issue;
+        const repository = event.cookedEvent.data.repository;
+        const ownerLogin = repository.owner.login;
+        const name = repository.name;
+        const rawbody = event.cookedEvent.data.issue.body;
+        const body = rawbody.replace(/:\w+:/, ''); // replace github icons :icon:
+
+        // Test each word for palindromeness
+        let palindromeCount: number = 0;
+        const bodyWords = body.match(/\b(\w+)\b/g);
+        _.forEach(bodyWords,
+          (bodyWord) => {
+            const lowerCaseWord = bodyWord.toLowerCase();
+            if (lowerCaseWord === lowerCaseWord.split('').reverse().join('')) {
+              palindromeCount += 1;
+            }
+        });
+
+        // Correctly pluralize the word palindrome
+        let countWord: string = 'palindromes';
+        if (palindromeCount === 1) {
+          countWord = 'palindrome';
+        }
+
+        // Test if the whole issue is a palindrome!
+        let palindromeCombo: boolean = false;
+        const bodySimplified = body.replace(/\W/g, '').toLowerCase();
+        if ( bodySimplified === bodySimplified.split('').reverse().join('') ) {
+          palindromeCombo = true;
+        }
```

Comment:
...which would then simplify this to just `const palindromeCombo = isPalidrome(body.replace(/\W/g, ''));`